core-java-modules/core-java-hex/src/test/java/com/baeldung/randomhexnumber/RandomHexGeneratorUnitTest.java

```diff
@@ -0,0 +1,121 @@
+package com.baeldung.randomhexnumber;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RandomHexGeneratorUnitTest {
+
+    private static RandomHexGenerator randomHexGenerator;
+
+    @BeforeAll
+    static void setUp() {
+        randomHexGenerator = new RandomHexGenerator();
+    }
+
+    @Test
+    void givenUsingRandomGenerator_whenGeneratingUnboundedHex_thenWillGetHexValue() {
```

Comment:
We could simplify test names by removing redundant givenUsingRandomGenerator 

`givenUsingRandomGenerator_whenGeneratingUnboundedHex_thenWillGetHexValue`

to

`givenGeneratingUnboundedHex_thenWillGetHexValue`